wgpu/src/lib.rs

```diff
@@ -1954,16 +1954,19 @@ impl Device {
     }
 
     /// Creates a shader module from either SPIR-V or WGSL source code.
```

Comment:
We should add a comment about what the result catches. We don't return an error on any validation problem, just parsing issues.

Reply:
I have trouble understanding how the error scopes and everything works... Why does `handle_error` exists and pushes to a some sink instead of just returning the error directly in the function ?
I understand that we want to be able to disable validation errors, but we could just return Ok in that case ?